boards/xtensa/esp32/doc/esp32.rst

```diff
@@ -0,0 +1,159 @@
+.. _esp32:
+
+ESP32
+#####
+
+Overview
+********
+
+From Wikipedia::
+
+>  ESP32 is a series of low cost, low power system on a chip microcontrollers
+>  with integrated Wi-Fi & dual-mode Bluetooth.  The ESP32 series employs a
+>  Tensilica Xtensa LX6 microprocessor in both dual-core and single-core
+>  variations.  ESP32 is created and developed by Espressif Systems, a
+>  Shanghai-based Chinese company, and is manufactured by TSMC using their 40nm
+>  process.
+
+The features include the following:
+
+- Dual core Xtensa microprocessor (LX6), running at 160 or 240MHz
+- 520KiB of SRAM
+- 802.11b/g/n/e/i
+- Bluetooth v4.2 BR/EDR and BLE
+- Various peripherals:
+  - 12-bit ADC with up to 18 channels
+  - 2x 8-bit DACs
+  - 10x touch sensors
+  - Temperature sensor
+  - 4x SPI
+  - 2x I2S
+  - 2x I2C
+  - 3x UART
+  - SD/SDIO/MMC host
+  - Slave (SDIO/SPI)
+  - Ethernet MAC
+  - CAN bus 2.0
+  - IR (RX/TX)
+  - Motor PWM
+  - LED PWM with up to 16 channels
+  - Hall effect sensor
+- Cryptographic hardware acceleration (RNG, ECC, RSA, SHA-2, AES)
+- 5uA deep sleep current
+
+System requirements
+*******************
+
+Prerequisites
+=============
+
+Two components are required in order to build this port: the `toolchain`_
+and the `SDK`_.  Both are provided by the SoC manufacturer.
+
+The SDK contains headers and a hardware abstraction layer library
+(provided only as object files) that are required for the port to
+function.
+
+The toolchain is available for Linux, Windows, and Mac hosts and 
+instructions on obtain and set them up are available in the ESP-IDF
+repository.
+
+Set up build environment
+========================
+
+With both the toolchain and SDK installed, the Zephyr build system
+must be instructed to use this particular variant by setting the
+```ZEPHYR_GCC_VARIANT```.  Two other environment variables should also
+be set, pointing to, respectively, the path where ESP-IDF can be
+located, and where the toolchain has been installed:
+
+.. code-block:: console
+
+   export ZEPHYR_GCC_VARIANT="espressif"
+   export ESP_IDF_PATH=/path/to/esp-idf
+   export ESPRESSIF_TOOLCHAIN_PATH=/path/to/xtensa-esp32-elf/
+
+Flashing
+========
+
+Issuing ```make BOARD=esp32 flash``` should work as usual.  Environment
+variables can be set to set the serial port device, baud rate, and
+other options.  Please refer to the following table for details.
+
++----------------+---------------+
+| Variable       | Default value |
++================+===============+
+| ESP_DEVICE     | /dev/ttyUSB0  |
+| ESP_BAUD_RATE  | 921600        |
```

Comment:
Need to put the row separator line in between each of these rows:

     | ESP_DEVICE     | /dev/ttyUSB0  |
     +----------------+---------------+
     | ESP_BAUD_RATE  | 921600        |